File: Year II/First Semester/MAP/Laborator/Laborator9/ConsoleApp1/ConsoleApp1/Model/SortingTask.cs
namespace ConsoleApp1.Model;

public class SortingTask: Task
{
    protected int[] Numbers;
    private AbstractSortingTask AbstractSortingTask { get; set; }

    public SortingTask(string taskId, string taskDescription, int[] numbers, AbstractSortingTask abstractSortingTask) : base(taskId, taskDescription)
    {
        Numbers = numbers;
        AbstractSortingTask = abstractSortingTask;
        this.AbstractSortingTask.Numbers = numbers;
    }

    public override void Execute()
    {
        try
        {
            this.AbstractSortingTask.Sort();
        }
        catch (Exception e)
        {
            Console.WriteLine(e);
        }
    }
    
}

File: Year II/First Semester/MAP/Laborator/Laborator2/model/SortingTask.java
package sem1_2.model;

public class SortingTask extends Task{
    int[] numbers;
    AbstractSorter sorter;
    SortingStrategy strategy;

    public SortingTask(String id, String description, int[] numbers, SortingStrategy strategy) {
        super(id, description);
        this.numbers = numbers;
        this.strategy = strategy;

        if(strategy == SortingStrategy.BUBBLE) {
            sorter = new BubbleSorter(numbers);
        }
        else sorter = new MergeSorter(numbers);
    }

    @Override
    public void execute(){
        sorter.sort();
        for(int number : numbers) {
            System.out.print(number + " ");
        }
        System.out.println();
    }
}
